bundle/compliance/policy/gcp/data_adapter.rego

```diff
@@ -7,31 +7,3 @@ resource = input.resource.resource
 iam_policy = input.resource.iam_policy
 
 has_policy = common.contains_key(input.resource, "iam_policy")
-
-is_storage_bucket {
-	input.subType == "gcp-storage-bucket"
-}
-
-is_iam_service_account {
-	input.subType == "gcp-iam-service-account"
-}
-
-is_iam_service_account_key {
-	input.subType == "gcp-iam-service-account-key"
-}
-
-is_cloudkms_crypto_key {
-	input.subType == "gcp-cloudkms-crypto-key"
-}
-
-is_bigquery_dataset {
-	input.subType == "gcp-bigquery-dataset"
-}
-
-is_bigquery_table {
-	input.subType == "gcp-bigquery-table"
-}
-
-is_compute_instance {
-	input.subType == "gcp-compute-instance"
-}
```

Comment:
using a set removes the repetitiveness in this file, so we only use the relevant strings twice:
1 - defined in the set `subtypes`
2 - the rule itself by calling `common.is_gcp_subtype`